### ISSUE TITLE
bug: Ionic components miss typings in Vue templates with Volar

### ISSUE THREAD

**Issue by elwinvaneede** [ionic-team/ionic-framework#24169]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

With [Volar](https://marketplace.visualstudio.com/items?itemName=johnsoncodehk.volar) you get the best TypeScript support for Vue 3 in VSCode (better than Vetur). Including templates.

For regular Vue 3 components this works: 

<img width="501" alt="Screenshot 2021-11-05 at 08 24 24" src="https://user-images.githubusercontent.com/35340250/140473521-e9f00061-5e01-4bcf-8100-b401263ac841.png">

But for Ionic Vue components this doesn't work:

<img width="459" alt="Screenshot 2021-11-05 at 08 27 05" src="https://user-images.githubusercontent.com/35340250/140473781-572d9677-67ad-47ec-a96a-ee6110c3671e.png">

### Expected Behavior

<img width="459" alt="Screenshot 2021-11-05 at 08 27 05" src="https://user-images.githubusercontent.com/35340250/140473781-572d9677-67ad-47ec-a96a-ee6110c3671e.png">

For the `collapse` prop of `ion-header` I would expect to see `collapse?: 'condense' | 'fade'` as the typing and get errors when I enter something else. Plus I would expect to see the docblock containing "Describes the scroll effect that will be applied to the header.".

I know I can use Vetur (which uses those Vetur json files), but Volar is much better for Vue 3 and has `<script setup>` support which I would like to start using.

### Steps to Reproduce

- Use Volar in VSCode (make sure Vetur is disabled)
- Navigate to a Ionic Vue project (see code reproduction url)
- Go to a `.vue` SFC file
- Hover over props of a Ionic component in the template
- Trigger autocomplete/intellisense

### Code Reproduction URL

https://github.com/elwinvaneede/ionic6-volar

### Ionic Info

```
Ionic:

   Ionic CLI : 6.16.3

Utility:

   cordova-res                          : 0.15.3
   native-run (update available: 1.4.1) : 1.3.0

System:

   NodeJS : v14.16.0
   npm    : 6.14.11
   OS     : macOS Monterey
```

### Additional Information

I think I've found the cause, through my work on the [Graphite Design System](https://github.com/paqtcom/graphite-design-system). But I might be wrong of course.

The `@ionic/vue` package has `vue: 3.0.0` in the package-lock.json, instead of the latest Vue 3 version (`3.2.21` at the time of writing).

When I use the latest vue version in `@graphite/vue` I do get the expected typing through Volar, but also a lot of false type errors, which makes it unusable in Volar:

<img width="824" alt="Screenshot 2021-11-05 at 08 49 10" src="https://user-images.githubusercontent.com/35340250/140476656-e106f18f-d2e7-4408-9ff5-2708d272370b.png">

<img width="824" alt="Screenshot 2021-11-05 at 08 49 18" src="https://user-images.githubusercontent.com/35340250/140476676-0a0134c2-da05-4d62-8036-e7e7bca7e6d4.png">

When I use exactly `vue: 3.0.0` in `@graphite/vue` the type errors go away so it makes it at least usable in Volar, but I also don't have any typings (just like Ionic Vue):

<img width="599" alt="Screenshot 2021-11-05 at 08 49 41" src="https://user-images.githubusercontent.com/35340250/140476705-8f6ea6c8-70fb-41c1-b04f-5e49199c33e2.png">

**Comment by elwinvaneede** [comment]
Fair enough to mark it as a feature request, I found it hard to determine if this is a bug or a feature request, so went with a bug. 

I would love this to be considered, since it greatly improves the DX (in my humble opinion), and a DX most Volar users might expect.

Either way, thanks for the great framework.

**Comment by elwinvaneede** [comment]
FYI: Visual Studio Code + Volar extension is the [recommended IDE setup in the Vue docs](https://vuejs.org/guide/quick-start.html#local).

**Comment by aaronksaunders** [comment]
wondering if this is going to be resolved? It does make it quite difficult to use with the latest ionic vue release

<img width="758" alt="Screen Shot 2022-06-21 at 6 45 19 PM" src="https://user-images.githubusercontent.com/524953/174909088-c410eb1d-7298-4b71-bf31-aa8ebdba26f1.png">


```
    "@ionic/vue": "6.1.10",
    "@ionic/vue-router": "6.1.10",
    "core-js": "^3.6.5",
    "vue": "3.2.37",
    "vue-router": "4.0.16"
```

**Comment by kachidk** [comment]
@aaronksaunders It's quite embarrassing I was trying to pitch to my team to pick up Ionic Vue and this happens.

**Comment by elwinvaneede** [comment]
I see the `@ionic/vue` package no longer uses `3.0.0` in their `package-lock.json` file. So this was to be expected based on my finding to be honest.

But it seems to be resolved through this more recent issue and will be available in the next release: https://github.com/ionic-team/ionic-framework/issues/25485 

CC: @liamdebeasi

**Comment by liamdebeasi** [comment]
Yes, the issue in https://github.com/ionic-team/ionic-framework/issues/24169#issuecomment-1162437607 should be resolved in Ionic 6.1.11.

**Comment by liamdebeasi** [comment]
Does this issue still reproduce in Ionic 6.1.11? I tested the sample repo in VS Code + Vetur and this appears to work now:

![image](https://user-images.githubusercontent.com/2721089/175056200-000da03c-3e6b-4ec5-a9a9-0ca22422c933.png)

**Comment by danielroe** [comment]
I can confirm this is working in [nuxt-ionic](https://github.com/danielroe/nuxt-ionic) with Volar.

**Comment by elwinvaneede** [comment]
I can also confirm it's working with Ionic 6.1.11. Thanks!
